mifospay/src/main/java/org/mifos/mobilewallet/mifospay/data/firebase/api/services/MifosPayInstanceIDService.java

```diff
@@ -44,6 +44,7 @@
     // [START refresh_token]
     @Override
     public void onTokenRefresh() {
+        if(mPreferencesHelper != null) {
```

Comment:
What if it will be `null`? Please explain.

Reply:
Simply it will throw Null Pointer exception when it is null. When the PreferenceHelper object is initialized it contains nothing and here we are calling setFireBaseRegId(refreshedToken) on that object. As it contains nothing, it is throwing Null pointer exception.

Reply:
This is incorrect.

The app crashes on first time since it is null and it doesn't crashes next time since `onTokenRefresh` is not even called. (Its only called when the token actually refreshes e.g. new install, app update etc).

See the comment in PR for more details